### PR TITLE
PDL recipe

### PR DESCRIPTION
**Pull request by vazirim**

### CODE REVIEW

**Review by deanwampler — CHANGES_REQUESTED** [review]
I like the notebook a lot, but the last example worked very poorly. See the long comment.

I have a number of wording suggestions that I'll make with PR to your branch. They include explaining why this notebook uses an extension and I "rephrased" the mention of LiveLLM, which we don't use anywhere else. 

I also cleared the output, which is our practice. In doing so, I noticed a strange behavior for the output of the chat. Invoking "Clear outputs of all cells" in Jupyter Lab, it only erased one line of the chat interaction. I had to repeat the command multiple times to clear the _whole_ output. This seems like a bug. I'll file an issue with your repo for this one.

File: recipes/PDL/pdl.ipynb [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+  {
+   "cell_type": "markdown",
+   "id": "50f4ce4a",
+   "metadata": {},
+   "source": [
+    "# Prompt Declaration Language\n",
+    "\n",
+    "Prompt engineering is difficult: minor variations in prompts have large impacts on the output of LLMs and prompts are model-dependent. In recent years <i> prompt programming languages </i> have emerged to bring discipline to prompt engineering. Many of them are embedded in an imperative language such as Python or TypeScript, making it difficult for users to directly interact with prompts and multi-turn LLM interactions.\n",
+    "\n",
+    "The Prompt Declaration Language (PDL) is a YAML-based declarative approach to prompt programming, where prompts are at the forefront. PDL facilitates model chaining and tool use, abstracting away the plumbing necessary for such compositions, enables type checking of the input and output of models, and is based on LiteLLM to support a variety of model providers. PDL has been used with RAG, CoT, ReAct, and an agent for solving SWE-bench. PDL is [open-source](https://github.com/IBM/prompt-declaration-language) and works well with watsonx.ai and Granite models.\n",
```
[the remark this answers is not in the document]
Comment (deanwampler) [reply recipes/PDL/pdl.ipynb:12]: I forgot to delete this comment, after discussions with Adam... Sorry.

**Review by rawkintrevo — CHANGES_REQUESTED** [review]
Need to add this to the testing suite.

.github/workflows/vanilla_workflow.yaml
.github/notebook_lists/vanilla_notebooks.txt

**Comment by rawkintrevo** [comment]
To clarify my last comment, you need to add the notebook to both files- the yaml says "fire this test if any of these files are changed", then later in the testing it loads the txt file and tests every file listed in there. 

@deanwampler specifying the files is problematic, because sometimes (in the case of ollama or the java ones) you add the path to a different yaml / text file. But changing the PR template is trivial, i can certainly add more information.

**Comment by deanwampler** [comment]
Thanks. This is pure Python, so I'll add the notebook to both files in my PR to @vazirim's PR branch.

**Comment by vazirim** [comment]
Hi @deanwampler, thanks for all your suggestions. I incorporated your branch in this one.

**Comment by deanwampler** [comment]
I have a theory for while the CI is failing. It's not this PR's notebook. I'm going to merge this PR shortly and do a separate PR for the final steps required.

**Pull request merged**